framework/src/meshgenerators/FileMeshGenerator.C

```diff
@@ -37,7 +42,25 @@ FileMeshGenerator::generate()
 {
   auto mesh = _mesh->buildMeshBaseObject();
 
-  mesh->read(_file_name);
+  bool exodus =
+      _file_name.rfind(".exd") < _file_name.size() || _file_name.rfind(".e") < _file_name.size();
+  bool has_exodus_integers = isParamValid("exodus_extra_element_integers");
+  if (exodus && has_exodus_integers)
```

Comment:
Is this the only way to add in this extra logic (i.e. can we not bake it in for other MeshGenerators or into the mesh base class automatically)? One of the issues as you are well aware is the scalability with having to do all the reading on one processor, etc. I'm willing to add this in as-is, but just curious if you've looked at making this more of a first-class capability.

Reply:
Unfortunately no. The loading of elemental variables has to happen together with loading the mesh data, otherwise the element could be renumbered and we loss the track what the element integers in the exodus file are for what elements. I thought of creating a dedicated mesh generator like `ExodusFileMeshGenerator` to keep this untouched. If you like this better, I can do it.

About the 'reading on one processor', the code here is the same as what in in `UnstructuredMesh::read()`.

Reply:
This will work for now.